src/test/java/io/openmessaging/storage/dledger/util/DLedgerUtilsTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.openmessaging.storage.dledger.util;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import io.openmessaging.storage.dledger.utils.DLedgerUtils;
+import java.io.IOException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public class DLedgerUtilsTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+  
+  @PrepareForTest({DLedgerUtils.class, System.class})
+  @Test
+  public void testComputNextTimeMillis() {
+    PowerMockito.mockStatic(System.class);
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(1515585600000L);
+
+    assertEquals(1515628800000L, DLedgerUtils.computNextMorningTimeMillis());
+    assertEquals(1515585660000L, DLedgerUtils.computNextMinutesTimeMillis());
+    assertEquals(1515589200000L, DLedgerUtils.computNextHourTimeMillis());
+    assertEquals(1515591000000L, DLedgerUtils.computNextHalfHourTimeMillis());
+  }
+
+  @Test
+  public void testCrc32() {
+    assertEquals(0, DLedgerUtils.crc32(null));
+    assertEquals(0, DLedgerUtils.crc32(new byte[]{}));
+  }
+
+  @Test
+  public void testBytes2string() {
+    assertEquals("98", DLedgerUtils.bytes2string(new byte[]{-104}));
+  }
+
+  @Test
+  public void testString2bytes() {
+    assertNull(DLedgerUtils.string2bytes(""));
+    assertArrayEquals(new byte[] {-1}, DLedgerUtils.string2bytes(")@"));
```

Comment:
This is unexpected/undefined behaviour - could add a test where the string is an even number of hex chars